### PR TITLE
WIP: Redesign time display

### PR DESCRIPTION
**Pull request by winniehell**
This creates a new design for the time display widget.

It is still work in progress because the code needs some cleanup before merging.

| Before | After |
| --- | --- |
| Bildschirmfoto vom 2019-08-10 23-57-26 | Bildschirmfoto vom 2019-08-10 23-55-35 |
| Bildschirmfoto vom 2019-08-10 23-57-31 | Bildschirmfoto vom 2019-08-10 23-55-40 |

video: 
[new-time-display.zip](https://github.com/LMMS/lmms/files/3489322/new-time-display.zip)

closes https://github.com/LMMS/lmms/issues/5097

### CODE REVIEW

File: src/gui/widgets/TimeDisplayWidget.cpp [hunk 1]
```diff
@@ -34,20 +34,42 @@
...
+	m_majorValue.setAlignment(Qt::AlignRight);
+	m_majorValue.setStyleSheet(valueStyle);
+	m_minorLabel.setAlignment(Qt::AlignRight);
+	m_minorLabel.setStyleSheet(labelStyle);
+	m_minorValue.setAlignment(Qt::AlignRight);
+	m_minorValue.setStyleSheet(valueStyle);
+	m_milliSecondsLabel.setAlignment(Qt::AlignRight);
+	m_milliSecondsLabel.setStyleSheet(labelStyle);
+	m_milliSecondsValue.setAlignment(Qt::AlignRight);
+	m_milliSecondsValue.setStyleSheet(valueStyle);
```
Comment (winniehell) [inline src/gui/widgets/TimeDisplayWidget.cpp]: This repetition deserves its own widget.

File: src/gui/widgets/TimeDisplayWidget.cpp [hunk 2]
```diff
@@ -34,20 +34,42 @@
...
-
-	setMaximumHeight( 32 );
+	m_spinBoxesLayout.addWidget( &m_majorLabel, 0, 0 );
+	m_spinBoxesLayout.addWidget( &m_majorValue, 1, 0 );
+	m_spinBoxesLayout.addWidget( &m_minorLabel, 0, 1 );
+	m_spinBoxesLayout.addWidget( &m_minorValue, 1, 1 );
+	m_spinBoxesLayout.addWidget( &m_milliSecondsLabel, 0, 2 );
+	m_spinBoxesLayout.addWidget( &m_milliSecondsValue, 1, 2 );
+
+	this->setStyleSheet( "TimeDisplayWidget { border-radius: 4px; background-color: rgba(0,0,0,70); margin: 8px; padding: 4px; }" );
```
Comment (winniehell) [inline src/gui/widgets/TimeDisplayWidget.cpp]: This should go in the stylesheet.

File: src/gui/widgets/TimeDisplayWidget.cpp [hunk 3]
```diff
@@ -34,20 +34,42 @@
...
+
+	QString labelStyle = "font-size: 10px;";
+	QString valueStyle = "color: #0bd556; font-size: 24px;";
```
Comment (winniehell) [inline src/gui/widgets/TimeDisplayWidget.cpp]: This should go in the stylesheet.

**Comment by qnebra** [comment]
In my opinion numbers should have same style, now your PR looks quite incoherently.

**Comment by winniehell** [comment]
@qnebra Yes, I agree! :+1: I will create another pull request to TEMPO/BPM and TIME SIG consistent after this design got approved.

**Comment by winniehell** [comment]
@Umcaruje given that you are the UI/UX maintainer, can I get your feedback here?

**Comment by Umcaruje** [comment]
In my opinion, the number size is a bit too large, especially if you compare it to the buttons on the left, so I would make it a bit smaller, something around 20px should be good

> I will create another pull request to TEMPO/BPM and TIME SIG consistent after this design got approved.

I would suggest you do this in this PR.

Also, I really liked the idea of either making the unused zeroes in the minutes darker, that you posted somewhere else.

Also I would return the CPU and waveform meter to its previous place. 

All in all, this looks more modern and I'm all for it, but the tempo and time signature should be integrated into the actual design, as this looks wildly inconsistent at the moment

**Comment by winniehell** [comment]
@Umcaruje Thank you for your feedback! :+1: 

> something around 20px should be good

sure, I will give that a try.

> I would suggest you do this in this PR.

My intention was to keep this one as small as possible code-wise. However I trust your judgement and will update the other two components as well.

> I really liked the idea of either making the unused zeroes in the minutes darker

Would you restrict that to minutes (and bars) only or should I apply that to all segments?

> Also I would return the CPU and waveform meter to its previous place.

Thank you for catching this! :heart: This was definitely not intended but I didn't notice. :see_no_evil: I will revert that change.

**Comment by Umcaruje** [comment]
>My intention was to keep this one as small as possible code-wise.

Yeah but in terms of design it will be much easier to fine tune it if it's in the same PR.

>Would you restrict that to minutes (and bars) only or should I apply that to all segments?

Only minutes and bars, it wouldn't make sense on the other values.

**Comment by PhysSong** [comment]
It seems like there's another work #5261 in progress.

**Comment by winniehell** [comment]
yep, thank you @PhysSong. I'm closing this one.

**Pull request closed without merging**